e2e/docker/swupdate/Dockerfile

```diff
@@ -1,78 +1,33 @@
-FROM ubuntu:noble AS build
+FROM alpine:3.22 AS build
 
-ENV DEBIAN_FRONTEND=noninteractive
+WORKDIR /app
 
-RUN mkdir -p /app && apt update && apt install -y build-essential \
-    automake \
-    cmake \
-    curl \
-    libzmq3-dev \
-    liblua5.2-dev \
-    libluajit-5.1-dev \
-    libconfig-dev \
-    libarchive-dev \
-    libbtrfsutil-dev \
-    libjson-c-dev \
-    libyaml-dev \
-    zlib1g-dev \
-    git \
-    uuid \
-    uuid-dev \
-    liblzo2-dev \
-    libsystemd-dev \
-    libsystemd0 \
-    check \
-    librsync2 \
-    librsync-dev \
-    libext2fs-dev \
-    liburiparser-dev \
-    doxygen \
-    graphviz \
-    autoconf-archive \
-    linux-headers-generic \
-    libmbedtls-dev \
-    libcmocka-dev \
-    libfdisk-dev \
-    libwebsockets-dev \
-    libgpiod-dev \
-    libcurl4-openssl-dev \
-    libpci-dev \
-    gawk \
-    cpio \
-    meson \
-    ninja-build \
-    libzstd-dev \
-    wget \
-    python3 \
-    libebgenv-dev \
-    libmtd-dev \
-    libubi-dev \
-    libubootenv-dev \
-    libzck-dev
+RUN apk add --no-cache alpine-sdk python3 cmake zlib-dev linux-headers bsd-compat-headers yaml-dev json-c-dev curl-dev libconfig-dev \
```

Comment:
I liked the previous style. Maybe in general, try to align the style with the other Dockerfiles (e.g. [Dockerfile](https://github.com/UpstreamDataInc/goosebit/blob/master/docker/Dockerfile)). However, considering this is only used for tests, I don't care that much.